packages/contracts/contracts/attester-utils/ReimburseAttestation.sol

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {Unirep} from '../Unirep.sol';
+import 'hardhat/console.sol';
+
+/**
+ * @title ReimburseAttestation
+ * @dev An attester can inherit this contract to hold budget and allow the callers of the contract (attester app users) to be reimbersed for attestations.
+ 	This would make the publicly funding operation of attesters more modular and transparent.
+	e.g. An attester could directly get funded from a gitcoin round for its operation.
+ */
+contract ReimburseAttestation {
+    bool public accpetDonations;
+
+    mapping(address => bool) public whitelist;
+
+    Unirep public unirep;
+
+    // id of the attester this budget contract is deployed for
+    uint256 public attesterId;
+
+    event Reimbursed(address indexed recepient, uint amount, string txType);
+    event FundsReceived(address sender, uint amount);
+
+    constructor(Unirep _unirep, uint256 _attesterId) {
+        unirep = _unirep;
+        attesterId = _attesterId;
+        accpetDonations = true;
+    }
+
+    /**
+     * @dev Add users to whitelist
+     */
+    function addToWhitelist(
+        address[] memory addresses,
+        address owner
+    ) public virtual {
+        require(
+            msg.sender == owner,
+            'Only owner contract itself can call addToWhitelist()'
+        );
+        for (uint i = 0; i < addresses.length; i++) {
+            whitelist[addresses[i]] = true;
+        }
+        console.log('addresses[0] ', addresses[0]);
+    }
+
+    /**
+     * @dev Remove users from whitelist
+     */
+    function removeFromWhitelist(
+        address[] calldata toRemoveAddresses,
+        address owner
+    ) external {
+        require(
+            msg.sender == owner,
+            'Only owner contract itself can callremoveFromWhitelist()'
+        );
+        for (uint i = 0; i < toRemoveAddresses.length; i++) {
+            delete whitelist[toRemoveAddresses[i]];
+        }
+    }
+
+    /**
+     * @dev The contract should hold the funds and accept funds from anyone.
+     */
+    function donate() public payable {
+        emit FundsReceived(msg.sender, msg.value);
+    }
+
+    // Fallback function to accept ether sent directly to the contract
+    receive() external payable {
+        emit FundsReceived(msg.sender, msg.value);
+    }
+
+    /**
+     * @dev Restrict/allow the contract to receive more donations.
+
+     */
+    function toggleAcceptingDonations() public {
+        require(
+            msg.sender == address(this),
```

Comment:
I think this is condition is impossible? Maybe allow the address `attesterId` to control this, or add an `owner` variable?